app/policies/application_policy.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class ApplicationPolicy
+  attr_reader :user, :record
+
+  def initialize(user, record)
+    @user = user
+    @record = record
+  end
+
+  def index?
```

Comment:
to being on the safe side, better don't define such methods in an abstract class

imagine, sb registered a new policy `BookPolicy < ApplicationPolicy` and forgot to redefine method `update`: everything would look usually, but action update wouldn't work(customer would be redirected to some another page) without any hint why did that happen. seeing default error `method :update? is not defined` is much better and already says what is wrong

ideally, you can do sth like that:
```
protected

def update?
  raise NotImplementedError, 'define method :update?'
end
```

Reply:
done